dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/constraints/DatadogDataConstraints.kt

```diff
@@ -67,6 +68,34 @@ internal class DatadogDataConstraints : DataConstraints {
         return convertedAttributes.take(MAX_ATTR_COUNT).toMap()
     }
 
+    override fun validateEvent(rumEvent: Any): Any {
+        return if (rumEvent is ViewEvent) {
+
+            val customTimings = rumEvent.view.customTimings?.let {
+                it.copy(
+                    additionalProperties = it.additionalProperties.mapKeys { entry ->
+                        val sanitizedKey =
+                            entry.key.replace(Regex("[^a-zA-Z0-9\\-_.@$]"), "_")
+                        if (sanitizedKey != entry.key) {
+                            devLogger.w(
```

Comment:
reminds me the following:

![image](https://user-images.githubusercontent.com/4046447/109025814-5dd4f280-76bf-11eb-945a-61029abcf0ee.png)

but this is because models are immutable. another option is to do the same with JSON, which can be mutated in-place, but then we lose support of the type system.